src/tests/src/solver/optim/api/testApiOrtoolsLinearProblem.cpp

```diff
@@ -28,101 +28,119 @@
 
 using namespace Antares::Solver::Optim;
 
-struct Fixture
+struct FixtureEmptyProblem
 {
-    ~Fixture() = default;
-
-    Fixture()
+    FixtureEmptyProblem()
     {
         pb = std::make_unique<OrtoolsImpl::OrtoolsLinearProblem>(false, "sirius");
     }
 
     std::unique_ptr<OrtoolsImpl::OrtoolsLinearProblem> pb;
-
-    void createProblemMaximize();
 };
 
-void Fixture::createProblemMaximize()
+struct FixtureInfeasibleProblem: public FixtureEmptyProblem
 {
-    auto* var1 = pb->addNumVariable(0, 1, "var1");
-    auto* var2 = pb->addNumVariable(0, 1, "var2");
-    auto* constraint = pb->addConstraint(1, 1, "constraint");
+    using FixtureEmptyProblem::FixtureEmptyProblem;
 
-    constraint->setCoefficient(var1, 1);
-    constraint->setCoefficient(var2, 1);
+    FixtureInfeasibleProblem()
+    {
+        auto* var = pb->addNumVariable(0, 1, "var");
+        auto* constraint = pb->addConstraint(2, 2, "constraint");
+        constraint->setCoefficient(var, 1);
+    }
+};
 
-    pb->setObjectiveCoefficient(var1, 1);
-    pb->setObjectiveCoefficient(var1, 1);
-    pb->setMaximization();
-}
+struct FixtureFeasibleProblem: public FixtureEmptyProblem
+{
+    using FixtureEmptyProblem::FixtureEmptyProblem;
 
-BOOST_AUTO_TEST_SUITE(optim_api)
+    FixtureFeasibleProblem()
+    {
+        auto* var = pb->addNumVariable(0, 10, "var");
+        auto* constraint = pb->addConstraint(1, 1, "constraint");
+        constraint->setCoefficient(var, 1);
+        pb->setObjectiveCoefficient(var, 1);
+    }
+};
+
+BOOST_AUTO_TEST_SUITE(tests_on_OrtoolsLinearProblem)
 
-BOOST_AUTO_TEST_CASE(solverMip)
+BOOST_FIXTURE_TEST_CASE(add_int_variable_to_problem___check_var_exists, FixtureEmptyProblem)
 {
-    auto pb = std::make_unique<OrtoolsImpl::OrtoolsLinearProblem>(true, "sirius");
+    pb->addIntVariable(0, 1, "var");
+    auto* var = pb->getVariable("var");
+    BOOST_CHECK(var);
```

Comment:
you can also check lb & ub since they are set by the addIntVariable method

Reply:
Done